In [1]:
import joblib

loaded_model = joblib.load('random_forest_model.joblib')

In [2]:
import pandas as pd
from sklearn.metrics import mean_squared_error, r2_score

# importing the validation set
X = pd.read_csv('X validation.csv')
y = pd.read_csv('y validation.csv')

# predicting the validation set
y_pred = loaded_model.predict(X)

# calculating the metrics
mse = mean_squared_error(y, y_pred)
r2 = r2_score(y, y_pred)

# printing the metrics
print(f"Mean Squared Error: {mse}")
print(f"R-squared: {r2}")

Mean Squared Error: 14.19917159978103
R-squared: 0.9513502305831497


### Summary of Model Predictions on Validation Set

1. **Model Used:**
   - The `RandomForestRegressor` model was used for predictions. This model was previously trained and saved using `joblib`.

2. **Validation Data:**
   - The validation set consists of features (`X`) and target values (`y`) loaded from CSV files.

3. **Prediction Process:**
   - The model was loaded from the file `random_forest_model.joblib`.
   - Predictions were made on the validation feature set (`X`).

4. **Performance Metrics:**
   - **Mean Squared Error (MSE):** The MSE on the validation set is 14.20, indicating the average squared difference between the predicted and actual values.
   - **R-squared (R²):** The R² score is 0.951, suggesting that the model explains 95.1% of the variance in the validation data.

5. **Conclusion:**
   - The `RandomForestRegressor` model performs well on the validation set, with a high R² score and relatively low MSE, indicating accurate predictions and good generalization to unseen data.


In [3]:
X['TARGET Prediction'] = y_pred
X['TARGET'] = y


In [11]:
# Saving Data to Database
from sqlalchemy import create_engine

host = r'127.0.0.1' # denotes that the db in a local installation
db = r'msds610' # db we just created
user = r'postgres' # using the postgres user for this demo
pw = r'Harideep' # this is the password established during installation
port = r'5432' # default port estabalished during install
schema = r'analytics' # schema we just created

# create a connection to the database
engine = create_engine(f'postgresql://{user}:{pw}@{host}:{port}/{db}')
connection = engine.connect()

In [12]:
X.to_sql('validation_predictions', engine, schema=schema, if_exists='replace', index=False)

# Verify the data was saved correctly
query = f"SELECT * FROM {schema}.validation_predictions LIMIT 5"
pd.read_sql(query, connection)




Delivery_person_Age  Delivery_person_Ratings  temperature  humidity  \
0                 31.0                      4.7        26.29      88.0   
1                 20.0                      4.7        24.91      43.0   
2                 29.0                      4.6        25.99      65.0   
3                 20.0                      4.7        19.07      47.0   
4                 37.0                      4.3        19.23      46.0   

   precipitation  Distance (km)  haversine_distance  Traffic_Density_Score  \
0            0.0           4.92        5.935821e-08                   4.92   
1            0.0          17.94        1.152805e-06                  53.82   
2            0.0          14.95        5.383444e-07                  44.85   
3            0.0          35.87        2.677049e-06                 143.48   
4            0.0          16.54        6.694473e-07                  49.62   

   Weather_Impact_Score  Experience_Adjusted_Ratings  ...  \
0                  88.0                        145.7  ...   
1                  45.0                         94.0  ...   
2                  70.0                        133.4  ...   
3                  49.0                         94.0  ...   
4                  48.0                        159.1  ...   

   weather_description_6  weather_description_7  weather_description_8  \
0                      0                      0                      0   
1                      0                      0                      0   
2                      0                      0                      0   
3                      0                      0                      0   
4                      0                      0                      0   

   weather_description_9  Traffic_Level_1  Traffic_Level_2  Traffic_Level_3  \
0                      0                1                0                0   
1                      0                0                0                1   
2                      0                0                0                1   
3                      0                0                0                0   
4                      0                0                0                1   

   Traffic_Level_4  TARGET Prediction     TARGET  
0                0          21.808000  23.333333  
1                0          39.533500  41.716667  
2                0          41.224167  42.533333  
3                1          63.431333  72.333333  
4                0          37.566500  36.283333  

[5 rows x 33 columns]